### PR TITLE
PR feedback from #418

### PR DESCRIPTION
**Pull request by Alan-Jowett**
PR feedback from #418 

Previous PR was closed before all feedback was fixed.

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

File: libs/platform/ebpf_platform.h [hunk 1]
```diff
@@ -258,11 +258,9 @@ extern "C"
 
     /**
      * @brief Query the platform for the total number of CPUs.
-     * @param[out] cpu_count Pointer to memory location that contains the
-     *    number of CPUs.
+     * @return The count of logical cores in the system.
      */
-    void
-    ebpf_get_cpu_count(_Out_ uint32_t* cpu_count);
+    _Ret_range_(>, 0) uint32_t ebpf_get_cpu_count();
```
Comment (dthaler) [inline libs/platform/ebpf_platform.h:263]: Also add same annotation to line 278

**Review by dthaler — APPROVED** [review]

File: libs/platform/ebpf_platform.h (hunk 1 above)
Comment (Alan-Jowett) [reply libs/platform/ebpf_platform.h:263]: I am not quite sure how to annotate ebpf_get_current_cpu return value as being in range [0, ebpf_get_cpu_count()).

It looks like the values need to be known at compile time.

Comment (dthaler) [reply libs/platform/ebpf_platform.h:263]: just say > 0

Comment (dthaler) [reply libs/platform/ebpf_platform.h:263]: Never mind, I misunderstood, 0 is legal.

**Pull request merged**